client/verta/verta/_tracking/deployable_entity.py

```diff
@@ -35,6 +47,167 @@ def _histogram_endpoint(self):
+        shutil.move(temp_path, path)
+
+        return path
+
+    def _get_cached_file(self, filename):
+        name, extension = os.path.splitext(filename)
+        if extension == '.zip':
+            pass
+        elif extension == '.tgz':
+            pass
+        elif extension == '.tar':
+            pass
+        elif extension == '.gz' and os.path.splitext(name)[1] == '.tar':
+            name = os.path.splitext(name)[0]
+        else:
+            name = filename
+
+        path = os.path.join(_CACHE_DIR, name)
+        return path if os.path.exists(path) else None
+
+    @abc.abstractmethod
+    def _get_artifact(self, key):
+        raise NotImplementedError
+
+    def fetch_artifacts(self, keys):
+        """
+        Downloads artifacts that are associated with a class model.
+
+        Parameters
+        ----------
+        keys : list of str
+            Keys of artifacts to download.
+
+        Returns
+        -------
+        dict of str to str
+            Map of artifacts' keys to their cache filepaths—for use as the ``artifacts`` parameter
+            to a Verta class model.
+
+        Examples
+        --------
+        .. code-block:: python
+
+            run.log_artifact("weights", open("weights.npz", 'rb'))
+            # upload complete (weights)
+            run.log_artifact("text_embeddings", open("embedding.csv", 'rb'))
+            # upload complete (text_embeddings)
+            artifact_keys = ["weights", "text_embeddings"]
+            artifacts = run.fetch_artifacts(artifact_keys)
+            artifacts
+            # {'weights': '/Users/convoliution/.verta/cache/artifacts/50a9726b3666d99aea8af006cf224a7637d0c0b5febb3b0051192ce1e8615f47/weights.npz',
+            #  'text_embeddings': '/Users/convoliution/.verta/cache/artifacts/2d2d1d809e9bce229f0a766126ae75df14cadd1e8f182561ceae5ad5457a3c38/embedding.csv'}
+            ModelClass(artifacts=artifacts).predict(["Good book.", "Bad book!"])
+            # [0.955998517288053, 0.09809996313422353]
+            run.log_model(ModelClass, artifacts=artifact_keys)
+            # upload complete (custom_modules.zip)
+            # upload complete (model.pkl)
+            # upload complete (model_api.json)
+
+        """
+        if not (isinstance(keys, list)
+                and all(isinstance(key, six.string_types) for key in keys)):
+            raise TypeError("`keys` must be list of str, not {}".format(type(keys)))
+
+        # validate that `keys` are actually logged
+        self._refresh_cache()
+        existing_artifact_keys = {artifact.key for artifact in self._msg.artifacts}
+        unlogged_artifact_keys = set(keys) - existing_artifact_keys
+        if unlogged_artifact_keys:
+            raise ValueError("`keys` contains keys that have not been logged: {}".format(sorted(unlogged_artifact_keys)))
+
+        # get artifact checksums
+        paths = {artifact.key: artifact.path
+                 for artifact in self._msg.artifacts}
```

Comment:
This was changed from
```python
        response = _utils.make_request("GET",
                                       "{}://{}/api/v1/modeldb/experiment-run/getArtifacts".format(self._conn.scheme, self._conn.socket),
                                       self._conn, params={'id': self.id})
        _utils.raise_for_http_error(response)
        paths = {artifact['key']: artifact['path']
                 for artifact in _utils.body_to_json(response)['artifacts']}
```
because thanks to caching on `self._msg`, we actually don't need to call an endpoint here at all.